### PR TITLE
Prepare for release

### PR DESCRIPTION
**Pull request by andreamari**
Description
-----------

Preparing for releasing Mitiq 0.16.0

License
-------

- [x] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

### CODE REVIEW

**Comment by andreamari** [comment]
Please don't merge since there are a couple of open PRs to be included in the change log.

**Review by natestemen — APPROVED** [review]
LGTM! :shipit:

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ Congratulations to the new member of the Mitiq team @natestemen and special than
 
 ### All Changes
 
+- Add section 2 (use case) on DDD to user guide (@nathanshammah, gh-1304)
+- Update DDD options documentation (third section) (andreamari, gh-1327)
```
Comment (natestemen) [inline CHANGELOG.md:16]: your handle isn't `@`'ed here.

**Review by natestemen — APPROVED** [review]
one comment, but not sure it really matters. LGTM.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,42 @@
...
+opening [issues](https://github.com/unitaryfund/mitiq/issues) on GihHub.
+A further notable addition is the function [generate_quantum_volume_circuit()](https://mitiq.readthedocs.io/en/latest/apidoc.html#mitiq.benchmarks.quantum_volume_circuits.generate_quantum_volume_circuit) by @nickdgardner, extending the Mitiq benchmarking module with quantum volume
+circuits.
+
+Congratulations to the new member of the Mitiq team @natestemen and special thanks to the external contributors @Aaron-Robertson, @nickdgardner, @ZhaoyiLi-HekJukZaaiZyuJan and @amirebrahimi!
+
+### All Changes
+
+- Add section 2 (use case) on DDD to user guide (@nathanshammah, gh-1304)
+- Update DDD options documentation (third section) (andreamari, gh-1327)
```
Comment (andreamari) [inline CHANGELOG.md:16]:
```suggestion
- Update DDD options documentation (third section) (@andreamari, gh-1327)
```

**Pull request merged**